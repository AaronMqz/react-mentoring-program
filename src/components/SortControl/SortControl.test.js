import React from "react";
import { render, screen, cleanup } from "@testing-library/react";
import SortControlComponent from "./SortControl";
import renderer from "react-test-renderer";

afterEach(cleanup);

describe("SortControl Tests:", () => {
  test("1. That component is visible.", () => {
    render(<SortControlComponent current={"title"} />);
    const sortBy = screen.getByText("TITLE");
    expect(sortBy).toBeInTheDocument();
  });

  it("2. Snapshot renders correctly", () => {
    const tree = renderer
      .create(<SortControlComponent current={"title"} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
