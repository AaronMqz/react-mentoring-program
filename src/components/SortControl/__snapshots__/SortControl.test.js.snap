// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SortControl Tests: 2. Snapshot renders correctly 1`] = `
<div
  className="sc-eDDNvR iCMmSC"
>
  <label
    className="sc-jTrPJq eRfCSB"
  >
    SORT BY
  </label>
  <div
    aria-label="current-sort"
    className="sc-beqWaB kCYvWq"
    onClick={[Function]}
  >
    <div
      className="sc-gueYoa kLKxyO"
    >
      TITLE
      <svg
        aria-hidden="true"
        className="svg-inline--fa fa-caret-down "
        data-icon="caret-down"
        data-prefix="fas"
        focusable="false"
        role="img"
        style={Object {}}
        viewBox="0 0 320 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M137.4 374.6c12.5 12.5 32.8 12.5 45.3 0l128-128c9.2-9.2 11.9-22.9 6.9-34.9s-16.6-19.8-29.6-19.8L32 192c-12.9 0-24.6 7.8-29.6 19.8s-2.2 25.7 6.9 34.9l128 128z"
          fill="currentColor"
          style={Object {}}
        />
      </svg>
    </div>
  </div>
</div>
`;
